### PR TITLE
Verifying the format of IDs and Keys

### PR DESCRIPTION
**Pull request by ftomassetti**

### CODE REVIEW

File: core/src/main/java/org/lionweb/lioncore/java/utils/ModelValidator.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     *
+     * Ids can only contain these symbols:
+     *
+     * - lowercase latin characters: a..z
+     * - uppercase latin characters: A..Z
+     * - arabic numerals: 0..9
+     * - underscore: _
+     * - hyphen: -
+     */
+    public static boolean isValidID(String id) {
```
Comment (dslmeinte) [inline core/src/main/java/org/lionweb/lioncore/java/utils/ModelValidator.java:24]: Can't this be a static method on some other class? Then the `ModelValidator` class is essentially superfluous.

File: core/src/main/java/org/lionweb/lioncore/java/utils/ValidationResult.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package org.lionweb.lioncore.java.utils;
```
Comment (dslmeinte) [inline core/src/main/java/org/lionweb/lioncore/java/utils/ValidationResult.java:1]: I think we envisioned that validation results/issues/etc. would also be a model, so we should define an M2 for this. That also means we can just reference the offending node through `id`.

File: core/src/main/java/org/lionweb/lioncore/java/utils/ModelValidator.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        if (id.length() == 0) {
+            return false;
+        }
+        for (char c : id.toCharArray()) {
```
Comment (dslmeinte) [inline core/src/main/java/org/lionweb/lioncore/java/utils/ModelValidator.java:28]: I think I'd prefer a proper regex for this, as that's generally more portable.

File: core/src/main/java/org/lionweb/lioncore/java/utils/ModelValidator.java (hunk 1 above)
Comment (ftomassetti) [reply core/src/main/java/org/lionweb/lioncore/java/utils/ModelValidator.java:24]: Ok, I will move it to MetamodelValidator and recreate ModelValidator later on

File: core/src/main/java/org/lionweb/lioncore/java/utils/ValidationResult.java (hunk 2 above)
Comment (ftomassetti) [reply core/src/main/java/org/lionweb/lioncore/java/utils/ValidationResult.java:1]: Right, but I think we did not yet define rules for that, so I would refactor this code once we get the validation m2

File: core/src/main/java/org/lionweb/lioncore/java/utils/ModelValidator.java (hunk 3 above)
Comment (ftomassetti) [reply core/src/main/java/org/lionweb/lioncore/java/utils/ModelValidator.java]: Changing it to a regex

**Pull request merged**